### DISCUSSION TITLE
How to build the event book on a local machine

### DISCUSSION THREAD

**Discussion by ifenty** [ECCO-Hackweek/ecco-2024#34] [Show and tell]
**To build a copy of the event book locally follow these instructions:**

1. install conda / mamba on your local machine
2. clone the [ecco-2024](https://github.com/ECCO-Hackweek/ecco-2024) repo
3. navigate to *ecco-2024/conda*
4. create the 'hackweek' conda environment: ```mamba create -f environment.yml```
5. navigate to *ecco-2024/scripts*
6. run ```./clean.sh; ./build_resources.sh```

after the build there will either have a success or failure message.  If failure, scroll up and try to identify the error message, fix your files, and try again.

**SUCCESS MESSAGE**
```
===============================================================================

Finished generating HTML for book.
Your book's HTML pages are here:
    book/_build/html/
You can look at your book by opening this file in a browser:
    book/_build/html/index.html
Or paste this line directly into your browser bar:
    file:///Users/ifenty/git_repo_others/ECCO-Hackweek/ecco-2024/book/_build/html/index.html            

```

**Comment by ifenty** [comment]
@ECCO-Hackweek/admins2024

**Comment by gaelforget** [comment]
does not work on my system. no idea why but here is the error message :

```
(base) ?130 conda % mamba create -f environment.yml
Traceback (most recent call last):
  File "/Users/gaelforget/miniforge3/bin/mamba", line 7, in <module>
    from mamba.mamba import main
  File "/Users/gaelforget/miniforge3/lib/python3.10/site-packages/mamba/mamba.py", line 18, in <module>
    from conda.cli.main import generate_parser, init_loggers
ImportError: cannot import name 'generate_parser' from 'conda.cli.main' (/Users/gaelforget/miniforge3/lib/python3.10/site-packages/conda/cli/main.py)
```

**Comment by gaelforget** [comment]
I also tried on the p-cluster just in case but no luck either. 

- `module load conda` fails
- `module load conda` fails
- dont know what else to try

**Comment by gaelforget** [comment]
I just tried building the book via GitHub action but that fails too. Not sure why

See https://github.com/gaelforget/ecco-2024/actions/runs/11248218135

**Reply by gaelforget** [reply]
> I just tried building the book via GitHub action but that fails too. Not sure why
> 
> See https://github.com/gaelforget/ecco-2024/actions/runs/11248218135

I take that back ; see https://gaelforget.github.io/ecco-2024/tutorials/Julia_setup_index.html
